src/main/java/com/example/weatherwaveapi/service/WeatherService.java

```diff
@@ -16,9 +34,53 @@ public class WeatherService {
     private String apiUrl;
     public RestTemplate restTemplate = new RestTemplate();
 
-    public Object getWeatherBySelectedCity(String city){
-        System.out.println(apiUrl);
-        String url = apiUrl + "?q=" + city + "&appid=" + apiKey;
-        return restTemplate.getForObject(url, Object.class);
+    public WeatherOpenApiContainer getWeatherBySelectedCity(String city) {
+        String url = getUrlForSelectedCity(city);
+        try {
+            ResponseEntity<WeatherOpenApiContainer> responseEntity = restTemplate.exchange(
+                    url,
+                    GET,
+                    new HttpEntity<>(new HttpHeaders()),
+                    new ParameterizedTypeReference<>() {
+                    }
+            );
+            return responseEntity.getBody();
+
+        } catch (HttpClientErrorException | HttpServerErrorException e) {
+            log.error("Exception: {}", e.getMessage());
```

Comment:
let's make this error message more meaningful, try to provide information during which process this error happened